framework/src/timesteppers/TimeStepper.C

```diff
@@ -165,10 +165,58 @@ TimeStepper::constrainStep(Real & dt)
 void
 TimeStepper::step()
 {
+  _converged = true;
+
+  if (!_executioner.legacyTimeExecution())
+  {
+    _fe_problem.execTransfers(EXEC_TIMESTEP_BEGIN);
+    if (!_fe_problem.execMultiApps(EXEC_TIMESTEP_BEGIN))
+    {
+      _converged = false;
+      _failure_count++;
+      _console << "Aborting as executing multiapps on timestep_begin failed" << std::endl;
+      return;
+    }
+
+    _fe_problem.execute(EXEC_TIMESTEP_BEGIN);
+    if (!_converged)
+    {
+      _failure_count++;
+      _console << "Aborting by a Terminator" << std::endl;
+      return;
+    }
+    _fe_problem.outputStep(EXEC_TIMESTEP_BEGIN);
+  }
```

Comment:
Yes, `EXEC_MULTIAPP_FIXED_POINT_BEGIN` and `EXEC_MULTIAPP_FIXED_POINT_END` do not make the cut if fixed point solve object is not the immediate inner solve object of the time stepper.

Reply:
I understand that. But the stepper's role should go more and more towards computing a time step and that's it. Having the stepper organize the solve is taking responsibilities from the executioner

Reply:
I think timestepper should be converted into a solve object in a long run. As its step function indicates, it is not just computing a time step. I do not know what design is in your mind, but we should have a solve object performing the time integration and stepping, the current timestepper is the closest to that solve object. This being said, I do not mind move the execution of timestep_begin and timestep_end into `Transient` executioner, but I think that is a wrong direction, because it is against our design of solving objects.